openfisca_core/parameters/vectorial_asof_date_parameter_node_at_instant.py

```diff
@@ -0,0 +1,66 @@
+import numpy
+
+from openfisca_core.parameters.parameter_node_at_instant import ParameterNodeAtInstant
+from openfisca_core.parameters.vectorial_parameter_node_at_instant import VectorialParameterNodeAtInstant
+
+
+class VectorialAsofDateParameterNodeAtInstant(VectorialParameterNodeAtInstant):
+    """
+    Parameter node of the legislation at a given instant which has been vectorized along some date.
+    Vectorized parameters allow requests such as parameters.housing_benefit[date], where date is a np.datetime64 type vector
+    """
+
+    @staticmethod
+    def build_from_node(node):
+        VectorialParameterNodeAtInstant.check_node_vectorisable(node)
+        subnodes_name = node._children.keys()
+        # Recursively vectorize the children of the node
+        vectorial_subnodes = tuple([
+            VectorialAsofDateParameterNodeAtInstant.build_from_node(node[subnode_name]).vector
+            if isinstance(node[subnode_name], ParameterNodeAtInstant)
+            else node[subnode_name]
+            for subnode_name in subnodes_name
+        ])
+        # A vectorial node is a wrapper around a numpy recarray
+        # We first build the recarray
+        recarray = numpy.array(
+            [vectorial_subnodes],
+            dtype=[
+                (subnode_name, subnode.dtype if isinstance(subnode, numpy.recarray) else 'float')
+                for (subnode_name, subnode) in zip(subnodes_name, vectorial_subnodes)
+            ]
+        )
+        return VectorialAsofDateParameterNodeAtInstant(node._name, recarray.view(numpy.recarray), node._instant_str)
+
+    def __getitem__(self, key):
+        # If the key is a string, just get the subnode
+        if isinstance(key, str):
+            key = numpy.array([key], dtype='datetime64[D]')
+            return self.__getattr__(key)
+        # If the key is a vector, e.g. ['1990-11-25', '1983-04-17', '1969-09-09']
+        elif isinstance(key, numpy.ndarray):
+            assert numpy.issubdtype(key.dtype, numpy.datetime64)
+            names = list(self.dtype.names)  # Get all the names of the subnodes, e.g. ['before_X', 'after_X', 'after_Y']
+            values = numpy.asarray([value for value in self.vector[0]])
+            names = [
+                name
+                for name in names
+                if not name.startswith("before")
+            ]
+            names = [
+                numpy.datetime64(
+                    "-".join(name[len("after_"):].split("_"))
+                )
+                for name in names
+            ]
+            conditions = sum([
```

Comment:
Je ne comprends pas pourquoi la somme du nombre d'éléments satisfaisants `name <= key` donne la position de  `result `dans `values ` ?

Reply:
On itère sur les names qui sont les dates `after_XXXX` (et la dimension courte, autant que de changement du paramères selon la date), en comparant avec `key` qui est un vecteur de dates (dimension longue, autant que d'individus) pour récupérer l'index de la valeur idoine càd celle en vigueur aux dates fournies par le vecteur key. 

Est-ce plus clair ?

Reply:
Yes thanks, sorry to forget to use English in Core.